### ISSUE TITLE
Analyzer fails to parse use of generalized function type

### ISSUE THREAD

**Issue by peter-ahe-google** [dart-lang/sdk#30124]
At f1f884011fec6b08874951cb617a706d132c150e, the analyzer fails to parse this:

```
class C {
  foo(List<String Function(String)> functions) => null;
}

main() {
  new C().foo(null);
}
```

**Comment by eernstg** [comment]
Right, the informal spec of generic function type aliases (which includes function types on the form `.. Function(..)`) is not yet LGTM'd (see https://codereview.chromium.org/2841483003/), but all recent versions of that document have agreed that function types can be used as actual type arguments. (_Generic_ function types are rejected as type arguments, but that's not a grammatical issue, it's a separate compile-time check). So the given example should be accepted.

**Comment by peter-ahe-google** [comment]
@eernstg I'm a bit surprised by this. This has already been implemented in Fasta and dart2js, and there are tests committed that use this.

**Comment by eernstg** [comment]
The [informal spec](https://github.com/dart-lang/sdk/blob/master/docs/language/informal/generic-function-type-alias.md) has been landed in commit 5904409d50fa9b800d015faadbe49cb13e358644.

**Comment by srawlins** [comment]
Analyzer now parses this.
